src/deluge/gui/ui/keyboard/keyboard_screen.cpp

```diff
@@ -153,7 +153,7 @@ ActionResult KeyboardScreen::padAction(int32_t x, int32_t y, int32_t velocity) {
 				pressedPads[idx].active = false;
 				markDead = idx;
 				if ((AudioEngine::audioSampleTimer - pressedPads[idx].timeLastPadPress) > FlashStorage::holdTime) {
-					pressedPads[idx].padPressHeld = true;
+					pressedPads[idx].padPressHeld = true; // why is this needed if the pad has been released?
```

Comment:
```suggestion
					pressedPads[idx].padPressHeld = true;
```

This logic detects when a pad in the `pressedPads` list has been held for long enough to be considered "held". That is, it detects the difference between a short tap and a long press.

Reply:
But isn't that already covered by the section above, in lines 126-133? If the pad is released, then it's not being held anymore, right?

Reply:
Oh I completely missed that this is in the release branch -- I think this is a copy/paste error and the check should be removed to unconditionally set `padPressHeld` to false here.